src/js/media-chrome-listbox.js

```diff
@@ -0,0 +1,416 @@
+    const listItemEls = Array.from(this.data).map((value) => {
+      const listItemEl = document.createElement(this.itemRenderer);
+      listItemEl.setAttribute('role', 'option');
+      listItemEl.id = value;
+      listItemEl.innerText = this.format(value);
+      if (value === selectedValue) {
+        this.selectedElement = listItemEl;
+      }
+      return listItemEl;
+    });
+    this.listboxElement.replaceChildren(...listItemEls);
+  }
+
+  /**
+   * @description
+   *  Register events for the listbox interactions
+   */
+  registerEvents() {
+    this.addEventListener('focus', this.handleFocus);
+    this.addEventListener('keydown', this.handleKeyPress);
+    this.listboxElement.addEventListener('click', this.handleItemClick);
+  }
+
+  handleFocus(evt) {
+    // this.listboxElement.focus();
+    this.activeElement = this.selectedElement ?? this.listItemElements[0];
+  }
+
+  /**
+   * @description
+   *  Check if an item is clicked on. If so, set as both activeElement and selectedElement.
+   * @param evt
+   *  The click event object
+   */
+  handleItemClick(evt) {
+    if (evt.target.getAttribute('role') !== 'option') return;
+    this.selectedElement = evt.target;
+    this.activeElement = evt.target;
+  }
+
+  /**
+   * Check if the selected option is in view, and scroll if not
+   */
+  updateScroll() {
+    const { activeElement } = this;
+    const { listboxElement: listbox } = this;
+    if (!activeElement || listbox.scrollHeight <= listbox.clientHeight) return;
+
+    const scrollBottom = listbox.clientHeight + listbox.scrollTop;
+    var elementBottom = activeElement.offsetTop + activeElement.offsetHeight;
+    if (elementBottom > scrollBottom) {
+      listbox.scrollTop = elementBottom - listbox.clientHeight;
+    } else if (activeElement.offsetTop < listbox.scrollTop) {
+      listbox.scrollTop = activeElement.offsetTop;
+    }
+  }
+
+  /**
+   * @description
+   *  Handle various keyboard controls; UP/DOWN will shift focus; SPACE selects
+   *  an item.
+   * @param evt
+   *  The keydown event object
+   */
+  handleKeyPress(evt) {
+    const { key } = evt;
+    const currentActiveIndex = this.activeIndex ?? -1;
+    const length = this.listItemElements.length;
+    switch (key) {
+      case 'ArrowUp': {
+        this.activeIndex = Math.max(currentActiveIndex - 1, 0);
+        evt.preventDefault();
+        break;
+      }
+      case 'ArrowDown': {
+        this.activeIndex = Math.min(currentActiveIndex + 1, length - 1);
+        evt.preventDefault();
+        break;
+      }
+      case 'Home':
+        evt.preventDefault();
+        this.activeIndex = 0;
+        break;
+      case 'End':
+        evt.preventDefault();
+        this.activeIndex = length - 1;
+        break;
+      case ' ':
+        evt.preventDefault();
+        this.selectedElement = this.activeElement;
+        break;
+      // fall through
+      default:
+        var itemToFocus = this.findItemToFocus(key);
+        if (itemToFocus) {
```

Comment:
this is to support search via typed string, recommended by WAI for lists.

Reply:
> I need...so much more context. Can you writeup an example guide for how a media chrome user would use this feature?

This is unlikely to be used directly by any non-advanced Media Chrome user. Rather, it'll be the core component for things like a settings overlay/dialog and captions flyaway menus (though there's a bit of discovery we'll have to do on whether or not we'll need to instrumentalize two different aria role configurations. It's possible we'd have a settings overlay/dialog implementation with slots that would result in less advanced users using this, but that's out of scope for this discussion imo.